fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/temp/view/RepeatedFieldSourceMatcher.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * RepeatedFieldSourceMatcher.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2019 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.temp.view;
+
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A Hamcrest matcher that checks whether a particular {@link Source} is a {@link RepeatedFieldSource}.
+ */
+public class RepeatedFieldSourceMatcher extends TypeSafeMatcher<Source> {
```

Comment:
I'm just going to pose this here, because I don't have a better idea, but have you given thought to how matchers are going to work in a multi-source world, i.e. differentiating between `SELECT r1.x, r2.y FROM type, type.repeated AS r1, type.repeated AS r2` vs `SELECT r1.x, r1.y FROM type, type.repeated AS r1`. Creating a matcher for this seems awfully close to creating a whole planner, so perhaps that is kind of pushing us towards making unit tests on rules, and then more functional tests that don't assert about the resulting plan, but instead on the results.